neuromation/api/plugins.py

```diff
@@ -0,0 +1,69 @@
+import numbers
+from typing import Any, Dict, List, Mapping, Tuple, Type, Union
+
+
+class ConfigBuilder:
+    _config_spec: Dict[str, Any]
+
+    def __init__(self) -> None:
+        self._config_spec = dict()
+
+    def _define_param(
+        self,
+        section: str,
+        name: str,
+        type: Union[
+            Type[bool],
+            Type[numbers.Real],
+            Type[numbers.Integral],
+            Type[str],
+            Tuple[Type[List[Any]], Type[bool]],
+            Tuple[Type[List[Any]], Type[str]],
+            Tuple[Type[List[Any]], Type[numbers.Real]],
+            Tuple[Type[List[Any]], Type[numbers.Integral]],
+        ],
+    ) -> None:
+        # Avoid circular import config.py <-> plugins.py
+        from .config import ConfigError
```

Comment:
We can move Exceptions to special file (like `exceptions.py`) to avoid problems like this, but it disagrees with "feature oriented" file structure.

Reply:
Doesn't matter really: the only public name should be `neuromation.api` anyway.
People should avoid importing nested modules.

A separate file for exceptions looks good in this case.

Reply:
I was more thinking about us, not public api users. I mean that it isn't very comfortable to have exception outside of the only file it is used. Also we cannot inforce this rule

Reply:
I don't insist, at least for now.